csrc/scheduler/mma_utils.h

```diff
@@ -300,6 +300,60 @@ std::pair<bool, bool> generateSharedMemoryEpilogueHeuristics(
     bool smem_b_reuse_guaranteed = false,
     bool ignore_occupancy_drop = false);
 
+struct MulSumAsMmaProps {
+  MulSumAsMmaProps(
+      BinaryOp* m,
+      ReductionOp* r,
+      TensorView* at = nullptr,
+      TensorView* bt = nullptr,
+      TensorView* out_t = nullptr,
+      BroadcastOp* bc_a = nullptr,
+      BroadcastOp* bc_b = nullptr)
+      : mop(m),
+        redop(r),
+        a(at),
+        b(bt),
+        out(out_t),
+        bcast_a(bc_a),
+        bcast_b(bc_b){};
+
+  BinaryOp* mop;
+  ReductionOp* redop;
+  TensorView* a;
+  TensorView* b;
+  TensorView* out;
+  BroadcastOp* bcast_a;
+  BroadcastOp* bcast_b;
+};
+
+//! Go through the fusion IR to find combinations of mul-sum
+//! which can be replaced with a mma op. This class operates
+//! in two phases. It can go through the graph and find the mul-sum
+//! pairs which can be replaced by a mma op. This phase returns a vector
+//! of properties of the mma op (MulSumAsMmaProps) which would replace the
+//! the mul-sum pair. It then exposes a function to replace with mma ops.
+class CombineMulSum : public IterVisitor {
+ public:
+  CombineMulSum(Fusion* fusion) : IterVisitor(), fusion_(fusion){};
+
+  //! Goes through the fusion to find mul-sum pairs.
+  std::vector<MulSumAsMmaProps> generateMulSumCanidates();
```

Comment:
This feels like it could be private.

Reply:
The plan is to use this in the mma scheduler in a follow-up diff. If I don't use it, I'll hide it.